src/modules/LanItems.rb

```diff
@@ -2698,9 +2698,18 @@ def export_udevs(devices)
           if !chan_ids["stdout"].empty?
             chanids = String.CutBlanks(Ops.get_string(chan_ids, "stdout", ""))
           end
+
           # we already know that kernel device exist, otherwise next above would apply
-          portname = ::File.read("/sys/class/net/#{device}/device/portname").strip
-          protocol = ::File.read("/sys/class/net/#{device}/device/protocol").strip
+          # FIXME: It seems that it is not always the case (bsc#1124002)
+          begin
+            portname = ::File.read("/sys/class/net/#{device}/device/portname").strip
+            protocol = ::File.read("/sys/class/net/#{device}/device/protocol").strip
```

Comment:
In the bug it fails when reading `/sys/class/net/eth0/device/protocol` file. That means the `/sys/class/net/eth0/device/portname` was read properly. So it seems only some files might be missing, not both at the same time.

So I'd prefer something like 
```ruby
            portname_file = "/sys/class/net/#{device}/device/portname"
            portname = File.read(portname_file).strip if File.exist?(portname_file)

            protocol_file = "/sys/class/net/#{device}/device/protocol"
            protocol = File.read(protocol_file).strip if File.exist?(protocol_file)
```

Reply:
Ah, that would result in `nil`, you need to do
```ruby
portname_file = "/sys/class/net/#{device}/device/portname"
portname = File.exist?(portname_file) ? File.read(portname_file).strip : ""
```

Reply:
Ok, so do you prefer to rescue the exception or just check if it exist before reading?

Reply:
For separate checks I'd prefer `File.exist?` as written above (to be consistent with the other YaST code). The exception would be better for the grouped case.